src/Document_Item.php

```diff
@@ -775,132 +658,73 @@ public static function showListForItem(CommonDBTM $item, $withtemplate = 0, $opt
-                    $canedit
-                    && ($withtemplate < 2)
-                ) {
-                    echo "<td width='10'>";
-                    Html::showMassiveActionCheckBox(__CLASS__, $data["assocID"]);
-                    echo "</td>";
-                }
-                echo "<td class='center'>$link</td>";
-                echo "<td class='center'>" . $data['entity'] . "</td>";
-                echo "<td class='center'>$downloadlink</td>";
-                echo "<td class='center'>";
-                if (!empty($data["link"])) {
-                    echo "<a target=_blank href='" . Toolbox::formatOutputWebLink($data["link"]) . "'>" . $data["link"];
-                    echo "</a>";
-                } else {
-                    echo "&nbsp;";
-                }
-                echo "</td>";
-                echo "<td class='center'>" . Dropdown::getDropdownName(
+            if (!isset($category_names[$data["documentcategories_id"]])) {
+                $category_names[$data["documentcategories_id"]] = Dropdown::getDropdownName(
                     "glpi_documentcategories",
                     $data["documentcategories_id"]
                 );
-                echo "</td>";
-                echo "<td class='center'>" . $data["mime"] . "</td>";
-                echo "<td class='center'>";
-                echo !empty($data["tag"]) ? Document::getImageTag($data["tag"]) : '';
-                echo "</td>";
-                echo "<td class='center'>" . Html::convDateTime($data["assocdate"]) . "</td>";
-                echo "</tr>";
-                $i++;
             }
-            echo $header_begin . $header_bottom . $header_end;
+            $entries[] = [
+                'itemtype' => self::class,
+                'row_class' => $data['is_deleted'] ? 'table-danger' : '',
+                'id'       => $data['assocID'],
+                'name'     => $name,
+                'entity'   => $data['entity'],
+                'filename' => $downloadlink,
+                'link'     => $link,
+                'headings' => $category_names[$data["documentcategories_id"]],
+                'mime'     => $data["mime"],
+                'tag'      => !empty($data["tag"]) ? Document::getImageTag($data["tag"]) : '',
+                'assocdate' => $data["assocdate"]
+            ];
         }
 
-        echo "</table>";
-        if ($canedit && $number && ($withtemplate < 2)) {
-            $massiveactionparams['ontop'] = false;
-            Html::showMassiveActions($massiveactionparams);
-            Html::closeForm();
-        }
-        echo "</div>";
+        TemplateRenderer::getInstance()->display('components/datatable.html.twig', [
+            'is_tab' => true,
+            'nopager' => true,
+            'nofilter' => true,
+            'columns' => $columns,
+            'formatters' => [
+                'name' => 'raw_html',
+                'filename' => 'raw_html',
+                'headings' => 'raw_html',
```

Comment:
The category name is a string to escape, but `link` column contains a `<a>` HTML tag that should be rendered without escaping.
```suggestion
                'link' => 'raw_html',
```

Reply:
Unless I miss something, `getDropdownName` will return HTML content only for the `comments` field if `$withcomment=true` or `&nbsp;` if `$default` is returned.

The `getDropdownName` method is used in both HTML and non-HTML context (e.g. data exports). IMHO, the returned `name` should only be raw data and we should always escape it when used in an HTML context. The `$default` param could be set to `''` instead of `'&nbsp;'`.